proof-systems/src/darlin/mod.rs

```diff
@@ -0,0 +1,223 @@
+//! The module for our Darlin proof carrying data (PCD) scheme as described in 
+//! our [DarlinProofTree doc](TODO: link). 
+//! The Darlin PCD scheme is based on (a variant of) Marlin/dlog, aggregating 
+//! the dlog "hard parts" as well as the inner sumchecks across multiple 
+//! circuits. 
+//! For now the module serves only basic structs and functions for the final 
+//! nodes of our conversion/exiting chain (which is either a "Simple Marlin", 
+//! or a "Final Darlin"). It is split into the following submodules
+//!     - `accumulators`: accumulator structs and their aggregation schemes as 
+//!     stand-alone non-interactive arguments. Although the stand-alone NI arguments
+//!     are not applied in recursion, they are useful for post-prossing.
+//!     - `pcd`: Proof carrying data from the verifier point of view. 
+//!     - `proof_aggregator`: utilities for proof post-processing, such as batch 
+//!     verification and aggregation of their dlog hard parts.
+pub mod pcd;
+pub mod accumulators;
+pub mod proof_aggregator;
+pub mod data_structures;
+pub mod error;
+
+pub mod tests;
+
+use algebra::{AffineCurve, ToConstraintField};
+use poly_commit::{ ipa_pc::{
+    UniversalParams, InnerProductArgPC,
+    CommitterKey as DLogProverKey,
+    VerifierKey as DLogVerifierKey,
+    Commitment
+}, PolynomialCommitment, QuerySet, LabeledCommitment, Evaluations};
+use marlin::{
+    Marlin,
+    ProverKey as MarlinProverKey, VerifierKey as MarlinVerifierKey,
+};
+use crate::darlin::{
+    data_structures::*,
+    pcd::{
+        PCD, PCDCircuit,
+        final_darlin::{FinalDarlinPCD, FinalDarlinPCDVerifierKey}
+    },
+    error::FinalDarlinError,
+};
+use rand::RngCore;
+use digest::Digest;
+use std::marker::PhantomData;
+
+
+/// FinalDarlin proof system. It is simply a (coboundary) Marlin SNARK of a dedicated
+/// recursive `PCDCircuit`.
```

Comment:
Why we don't we put the FinalDarlin proof system, even if dummy for now,  in an extra folder?

Reply:
It's the only one right now, in the future for sure